clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/DeployKubernetesAtomicOperation.groovy

```diff
@@ -59,10 +60,19 @@ class DeployKubernetesAtomicOperation implements AtomicOperation<DeploymentResul
   ReplicationController deployDescription() {
     task.updateStatus BASE_PHASE, "Initializing creation of replication controller."
 
+
+    task.updateStatus BASE_PHASE, "Looking up provided namespace..."
+
     def credentials = description.kubernetesCredentials
+    def namespace = description.namespace ? description.namespace : "default"
+
+    if (!credentials.isRegisteredNamespace(namespace)) {
+      throw new KubernetesIllegalArgumentException("Namespace $namespace was not provided as a registered namespace within Clouddriver. Registered namespaces are ${credentials.getNamespaces()}.")
```

Comment:
The request fails, which is what I want to have happen; however, I agree it might not be obvious to someone new to the API. The motivation was: I wanted to imply `default` only when no namespace was declared. This applies to both the credentials setup and the incoming requests. This seems reasonable, since this is generally how a "default" is handled. However, I also wanted to allow users to specifically restrict access to `default`, so requests that either imply or explicitly require creating a replication controller in `default` fail when they need to.